spec/lib/api/redfish-1.0/system-spec.js

```diff
@@ -866,140 +869,159 @@ describe('Redfish Systems Root', function () {
             });
     });
 
-    it('should return a valid system', function() {
-        redfish.getVendorNameById.resolves({
-            node: node,
-            vendor: 'notDellAndCisco'
+    describe('Get Systems By Indentifier', function () {
+        beforeEach('set up mocks', function(){
+            this.sandbox.stub(redfish, 'getVendorNameById');
         });
-        waterline.catalogs.findLatestCatalogOfSource.resolves(Promise.resolve({
-            node: node.id,
-            source: 'dummysource',
-            data: catalogData
-        }));
+        it('should return a valid system', function() {
+            redfish.getVendorNameById.resolves({
+                node: node,
+                vendor: 'undefined'
+            });
+            waterline.catalogs.findLatestCatalogOfSource.resolves(Promise.resolve({
```

Comment:
Using the object directly as the parameter for "resolves" is enough. Promise.resolve is not necessary here. Same as the following case. Just a suggestion for further work.